nimterop/globals.nim

```diff
@@ -51,15 +51,15 @@ type
     nocache*, debug*, past*, preprocess*, pnim*, pretty*, recurse*: bool
 
     consts*, enums*, procs*, types*: HashSet[string]
-
-    code*, constStr*, currentHeader*, debugStr*, enumStr*, mode*, procStr*, typeStr*: string
-    sourceFile*: string # eg, C or C++ source or header file
+    constStr*, debugStr*, enumStr*, procStr*, typeStr*: string
+    code*, currentHeader*, mode*, pluginFile*, sourceFile*: string
 
     ast*: Table[string, seq[ref Ast]]
     data*: seq[tuple[name, val: string]]
     when not declared(CIMPORT):
       grammar*: seq[tuple[grammar: string, call: proc(ast: ref Ast, node: TSNode) {.nimcall.}]]
 
+    onSymbol*: proc(sym: string): string {.cdecl.}
```

Comment:
in subsequent PR we shd the more flexible API as i had done here https://github.com/genotrance/nimterop/pull/65/files#diff-53f6738ffd5c92067bfb867ae8841e3a

```
proc onSymbol(result: var Result, symbol: Symbol, dummy: int) {.exportc.} =
=>
proc onSymbol(result: var Result, symbol: Symbol, dummy: int) {.exportc.} =
```

we'll definitely wanna augment what's passed as input and output and having those 2 object types avoids breaking APIs each time u wanna add a field

Reply:
I'm working on this.